src/App.tsx

```diff
@@ -1,26 +1,214 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getTodos, patchTodos, postTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Tabs } from './types/Tabs';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [input, setInput] = useState('');
+  const [formKey, setFormKey] = useState(0);
+  const [selectedTab, setSelectedTab] = useState(Tabs.All);
+  const [error, setError] = useState('');
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const [isLoading, setIsLoading] = useState<number[]>([]); // ids of the changed todos
+
+  const areAllCompleted = todos.every(todo => todo.completed);
+  const todoFocus: React.RefObject<HTMLInputElement> = useRef(null);
+
+  useEffect(() => {
+    async function getFetchedTodos() {
+      try {
+        const fetchedTodos = await getTodos();
+
+        setTodos(fetchedTodos);
+      } catch {
+        setError('Unable to load todos');
+        setIsErrorShown(true);
+      }
+    }
+
+    getFetchedTodos();
+  }, []);
+
+  useEffect(() => setFilteredTodos(todos), [todos]);
```

Comment:
no need to create a separate state for `filteredTodos` (if it depends on other state or props values).
Do something like this instead:
```suggestion
  const filteredTodos = useMemo(() => todos.filter(//some filter logic based on `selectedTab`), [todos]);
```